### ISSUE TITLE
Warning: Each child in a list should have a unique "key" prop.

### ISSUE THREAD

**Issue by lslee714** [Tintef/react-google-places-autocomplete#89]
Getting a `Warning: Each child in a list should have a unique "key" prop.` error when using this library inside a React Typescript application.

To reproduce:
- Inside a react with typescript app, initialize GooglePlacesAutocomplete with most basic configurations
- Start typing the ABCs, around 5-10 characters, this warning floods the console.

I probably could fix it and make a PR, but I thought I should submit the issue first.

**Comment by Tintef** [comment]
@lslee714 That's super weird 🤔 The only list is the suggestions list and it's actually using the suggestion id as the `key`: check it [here](https://github.com/Tintef/react-google-places-autocomplete/blob/4886c16b529f9371378f4386f8a2ac7d112720d2/src/GooglePlacesAutocomplete/index.js#L304)

I'm wondering what were the ids of the suggestions returned from that autocomplete call 🤔 

But yeah, if you figured out what happened there, a PR is more than welcomed!!

**Comment by lslee714** [comment]
Yeah it's weird, the error only happens once and after it happens, it doesn't complain again. I'll take a look sometime in the near future haha

**Comment by bencehornak** [comment]
I think the problem is on Google's side. Sometimes their API omits the id from the returning JSON for no reason :grimacing: 

For example, I intercepted this URL in the DevConsole:
```
https://maps.googleapis.com/maps/api/place/js/AutocompletionService.GetPredictionsJson?1sab&4sde-DE&7scountry%3AHU&9saddress&15e3&21m1&2e1&callback=_xdc_._8umyg&key=<My API KEY>&token=93086
```

If I fetch it repeatedly, sometimes `id` is included, sometimes it's not :unamused:

Good response:
```js
/**/_xdc_._8umyg && _xdc_._8umyg( {
   "predictions" : [
      {
         "description" : "Budapest, Abonyi utca, Ungarn",
         "id" : "5695d25a61ef2c2d780f2ae58e71fdde88c1c643", // here we have the id
         "matched_substrings" : [
            {
               "length" : 2,
               "offset" : 10
            }
         ],
         "place_id" : "Eh1CdWRhcGVzdCwgQWJvbnlpIHV0Y2EsIFVuZ2FybiIuKiwKFAoSCcNfVy6H3EFHEZVUoNFY40HvEhQKEgnJz9TRNMNBRxFgER4MKcQABA",
         "reference" : "Eh1CdWRhcGVzdCwgQWJvbnlpIHV0Y2EsIFVuZ2FybiIuKiwKFAoSCcNfVy6H3EFHEZVUoNFY40HvEhQKEgnJz9TRNMNBRxFgER4MKcQABA",
         "structured_formatting" : {
            "main_text" : "Abonyi utca",
            "main_text_matched_substrings" : [
               {
                  "length" : 2,
                  "offset" : 0
               }
            ],
            "secondary_text" : "Budapest, Ungarn"
         },
         "terms" : [
            {
               "offset" : 0,
               "value" : "Budapest"
            },
            {
               "offset" : 10,
               "value" : "Abonyi utca"
            },
            {
               "offset" : 23,
               "value" : "Ungarn"
            }
         ],
         "types" : [ "route", "geocode" ]
      },
...
```

Bad one:
```js
/**/_xdc_._8umyg && _xdc_._8umyg( {
   "predictions" : [
      {
         "description" : "Budapest, Abonyi utca, Ungarn", // id is missing !!!!!!!!!
         "matched_substrings" : [
            {
               "length" : 2,
               "offset" : 10
            }
         ],
         "place_id" : "Eh1CdWRhcGVzdCwgQWJvbnlpIHV0Y2EsIFVuZ2FybiIuKiwKFAoSCcNfVy6H3EFHEZVUoNFY40HvEhQKEgnJz9TRNMNBRxFgER4MKcQABA",
         "reference" : "Eh1CdWRhcGVzdCwgQWJvbnlpIHV0Y2EsIFVuZ2FybiIuKiwKFAoSCcNfVy6H3EFHEZVUoNFY40HvEhQKEgnJz9TRNMNBRxFgER4MKcQABA",
         "structured_formatting" : {
            "main_text" : "Abonyi utca",
            "main_text_matched_substrings" : [
               {
                  "length" : 2,
                  "offset" : 0
               }
            ],
            "secondary_text" : "Budapest, Ungarn"
         },
         "terms" : [
            {
               "offset" : 0,
               "value" : "Budapest"
            },
            {
               "offset" : 10,
               "value" : "Abonyi utca"
            },
            {
               "offset" : 23,
               "value" : "Ungarn"
            }
         ],
         "types" : [ "route", "geocode" ]
      },
...
```

**Comment by bencehornak** [comment]
Until it gets fixed, changing the `key` from `suggestion.id` to `suggestion.place_id` [here](https://github.com/Tintef/react-google-places-autocomplete/blob/4886c16b529f9371378f4386f8a2ac7d112720d2/src/GooglePlacesAutocomplete/index.js#L304) is a workaround

**Comment by Tintef** [comment]
@bencehornak Awesome, thanks for that research!

I think using the `place_id` is a good workaround and will add it to next minor realase

**Comment by lslee714** [comment]
@bencehornak Thanks for finding that! Haven't had a chance to investigate, thanks @Tintef for the update on the plans to fix it.

**Comment by Tintef** [comment]
@lslee714 released on version 2.2.2 🥳
